homeassistant/components/simplepush/notify.py

```diff
@@ -44,45 +54,120 @@ async def async_get_service(
         """Send a message to a Simplepush user."""
         title = kwargs.get(ATTR_TITLE, ATTR_TITLE_DEFAULT)
 
+        actions = None
+        action_ids = {}
+        attachments = None
         # event can now be passed in the service data
         event = None
+        feedback_action_timeout = None
+
         if data := kwargs.get(ATTR_DATA):
             event = data.get(ATTR_EVENT)
 
+            actions_data = data.get(ATTR_ACTIONS)
+            if isinstance(actions_data, list) and actions_data:
+                actions = []
+                for action in actions_data:
+                    if "action" in action and "url" in action:
+                        actions.append({"name": action["action"], "url": action["url"]})
+                    elif "action" in action:
+                        actions.append(action["action"])
+                        if "id" in action:
+                            action_ids.update({action["action"]: action["id"]})
+
+                try:
+                    feedback_action_timeout = int(
+                        data.get(ATTR_FEEDBACK_ACTION_TIMEOUT)
+                    )
+                except (ValueError, TypeError):
+                    feedback_action_timeout = 60
+
+            attachments_data = data.get(ATTR_ATTACHMENTS)
+            if isinstance(attachments_data, list) and attachments_data:
+                attachments = []
+                for attachment in attachments_data:
+                    if "attachment" in attachment and "thumbnail" in attachment:
+                        attachments.append(
+                            {
+                                "video": attachment["attachment"],
+                                "thumbnail": attachment["thumbnail"],
+                            }
+                        )
+                    elif "attachment" in attachment:
+                        attachments.append(attachment["attachment"])
+
         # use event from config until YAML config is removed
         event = event or self._event
 
+        def feedback_action_callback(
+            action_selected, action_selected_at, action_delivered_at, feedback_id
+        ):
+            payload = {
+                "action_selected": action_selected,
+                "action_selected_at": action_selected_at,
+                "action_delivered_at": action_delivered_at,
+                "feedback_id": feedback_id,
+            }
+
+            if action_selected in action_ids:
+                payload.update({"id": action_ids[action_selected]})
+
+            self.hass.bus.async_fire(EVENT_ACTION_TRIGGERED, payload)
+
         try:
             if self._password:
-                send(
-                    key=self._device_key,
-                    password=self._password,
-                    salt=self._salt,
-                    title=title,
-                    message=message,
-                    event=event,
+                self.hass.async_create_task(
```

Comment:
If we create a task the exceptions won't be caught here. We should await here.

Reply:
Hmm, this is a bit tricky.
If I don't create a task and use `await` instead, then a `wait_for_trigger` (see the example I provided in the description) will never be triggered.
And the possibility to use `wait_for_trigger` is crucial for this feature.

Do you have an idea on how to resolve this?